packages/block-library/src/block/test/edit.native.js

```diff
@@ -0,0 +1,111 @@
+/**
+ * External dependencies
+ */
+import { cleanup, fireEvent, waitFor } from '@testing-library/react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { getBlockTypes, unregisterBlockType } from '@wordpress/blocks';
+import { fetchRequest } from '@wordpress/react-native-bridge';
+
+/**
+ * Internal dependencies
+ */
+import { initializeEditor } from 'test/helpers';
+import { getReusableBlock } from './fixture';
+
+beforeAll( () => {
+	// Setup API fetch as it's done when initializing the editor
+	require( '@wordpress/react-native-editor/src/api-fetch-setup' ).default();
```

Comment:
Also, is there a reason to leverage inline `require` in some places rather than `import` atop the file?

Reply:
>Is there any part of this that we should mock (e.g. fetch)? Are the tests attempting to send actual requests?

Yes, the reusable block requires fetching its content from the site, that's why I had to mock the fetch requests.

>Also, is there a reason to leverage inline require in some places rather than import atop the file?

Not really, I think we could import it atop as regular imports.

I'd like to note that `api-fetch-setup` is not exported within the package `@wordpress/react-native-editor`, looks like it's meant to be internal because it's only used when initializing the editor.

Reply:
> Yes, the reusable block requires fetching its content from the site, that's why I had to mock the fetch requests.

Makes sense. 🤦🏻 I overlooked the fact that `@wordpress/react-native-bridge` is mocked within the Jest setup file. That is why I was confused. 

> I'd like to note that `api-fetch-setup` is not exported within the package `@wordpress/react-native-editor`, looks like it's meant to be internal because it's only used when initializing the editor.

Ah, interesting. I did not notice that until you pointed it out. I wonder if it makes sense to mock something higher up that does not require this internal reference. E.g. could/should we mock `@wordpress/api-fetch` and updated our assertion to check its mocked calls?

Reply:
> Technically we could mock `@wordpress/api-fetch` or set a custom handler as we do in [`api-fetch-setup.js`](https://github.com/WordPress/gutenberg/blob/f10179b9aa025aaf3513871c051998ab89abc993/packages/react-native-editor/src/api-fetch-setup.js#L53-L55). The only downside I see is that we won't take into account the extra logic we have in `packages/react-native-editor/src/api-fetch-setup.js` but that should be tested with unit tests.

Agreed. Testing the extra logic either within integration tests (like your current setup) or separately with unit tests is likely worthwhile. I don't have a strong opinion on which approach would be best at this point. I believe your current approach is sound, if you'd like to keep it in place. The import and initialization of the private/internal module is just a bit odd and unfortunate in my opinion.

Reply:
I'm going to try to mock the `@wordpress/api-fetch` package, if I don't find any blocker I rather have this approach, I'll let you know with the results 👍 .

Reply:
I mocked the `@wordpress/api-fetch` package and it worked 🎊 so I removed this part in this [commit](https://github.com/WordPress/gutenberg/pull/32038/commits/e99d7cf2a7c577596171f77b6c658ea6dd2581d8).